src/pages/index.js

```diff
@@ -1,84 +1,188 @@
-});
+
+const popupEditForm = new PopupWithForm(
+  "#profile-edit-modal",
+  ({ name, about }) => {
+    popupEditForm.saving(true);
+    return api.editProfile({ name, about }).then((editProfile) => {
+      popupEditForm.saving(false);
+      userInformation.setUserInfo(editProfile);
+      popupEditForm.close();
+    });
+  }
+);
 popupEditForm.setEventListeners();
+
 // popup with image
-const popupImage = new PopupWithImage("#preview-image-modal");
-popupImage.setEventListeners();
 
-// section
-function createCard(item) {
-  const cardElement = new Card(item, "#card-template", handleImageClick);
-  return cardElement.getView();
+function handleImageClick(data) {
+  const popupImage = new PopupWithImage("#preview-image-modal");
+  popupImage.setEventListeners();
+  popupImage.open(data);
 }
-
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: (item) => {
-      const cardElement = createCard(item);
-      section.addItem(cardElement);
-    },
-  },
-  ".cards__list"
-);
-section.renderItems();
-
 // functions
 
-function handleCardFormSubmit(data) {
-  const cardInput = createCard(data);
-  section.addItem(cardInput);
-  newCardPopup.close();
+function createCard(item, _id) {
+  const cardElement = new Card(
+    item,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCardClick,
+    handleImageLike
+  );
+  return cardElement.getView();
 }
 
-function handleImageClick(data) {
-  popupImage.open(data);
+function handleCardAddFormSubmit(data) {
+  newCardPopup.saving(true);
+  api.postNewCard(data).then((card) => {
+    newCardPopup.saving(false);
+    const cardInput = createCard(card);
+    section.addItem(cardInput);
+    newCardPopup.close();
+  });
 }
 
-const editFormValidator = new FormValidation(
-  validationSettings,
-  DOM.profileEditForm
-);
-editFormValidator.enableValidation();
-
-const addFormValidator = new FormValidation(
-  validationSettings,
-  DOM.profileAddForm
-);
-addFormValidator.enableValidation();
+enableValidation(validationSettings);
 
 //event listeners
 DOM.profileEditButton.addEventListener("click", () => {
-  const formData = userInfo.getUserInfo();
+  const formData = userInformation.getUserInfo();
   popupEditForm.setInputValues(formData);
   popupEditForm.open();
 });
 
 // add new card button
 DOM.profileAddButton.addEventListener("click", () => {
-  addFormValidator.toggleButtonState();
   newCardPopup.open();
 });
+
+// avatar image update button
+const popupAvatar = new PopupWithForm("#update-avatar-modal", (formData) => {
+  popupAvatar.saving(true);
+  const avatar = formData.avatar;
+  return api.updateAvatar(avatar).then((updateAvatar) => {
+    popupAvatar.saving(false);
+    userInformation.setAvatar(updateAvatar.avatar);
+    popupAvatar.close();
+  });
+});
+popupAvatar.setEventListeners();
+
+DOM.avatarImgButton.addEventListener("click", () => {
+  const formData = userInformation.getUserInfo();
+  formValidators["modal-avatar-form"].enableValidation();
```

Comment:
`enableValidation` should be called only 1 time for each instance of `FormValidator` to enable validation. You should not do that when you open/close a popup because you add more and more the same listeners to the same forms. This can cause a `memory leak`